tests/unit/editors/Bugs/tests.html

```diff
@@ -1289,6 +1301,33 @@
 				equal(editor.igPercentEditor("value"), 0.029, "value should be 0.029"); //fails prior fix and returns 29 instead
 					$("#695Editor").remove();
 			});
+
+				testId = 'Bug 809 - Wrong value is set when we have isLimitedToListValues: true and revertIfNotValid: false';
+				test(testId, 3, function () {
+				var editor =  $("#809Editor");
+				input = editor.igNumericEditor("field"),
+				editor.select();
+				keyInteraction(38, input); //Arrow Up
+				input.val("");
+				editor.blur();
+				equal(editor.igNumericEditor("value"), "", "Value should be empty string");
+				editor.igNumericEditor("option","allowNullValue", "true");
+				editor.select();
+				keyInteraction(38, input); //Arrow Up
+				input.val(null);
+				editor.blur();
+				equal(editor.igNumericEditor("value"), null, "Value should be null when allowNullValue=true");
+				editor.igNumericEditor("option","allowNullValue", "false");
+				editor.igNumericEditor("option","revertIfNotValid", "true");
+				editor.select();
+				keyInteraction(38, input); //Arrow Up
+				editor.blur();
+				editor.select();
+				keyInteraction(55, input); //7
```

Comment:
This should probably actually set the edit field value as it triggers the key events, otherwise there's no value to revert when the blur hits.